scvid/data/read.py

```diff
@@ -0,0 +1,43 @@
+import re
+
+from anndata import AnnData, read_h5ad
+from google.cloud import storage
+
+
+def read_h5ad_gcs(filename: str) -> AnnData:
+    r"""
+    Read `.h5ad`-formatted hdf5 file from the Google Cloud Storage.
+
+    Example::
+
+        >>> adata = read_h5ad_gcs("gs://dsp-cell-annotation-service/benchmark_v1/benchmark_v1.000.h5ad")
+
+    Args:
+        filename (str): Path to the data file in Cloud Storage.
+    """
+    assert filename.startswith("gs:")
+    # parse bucket and blob names from the filename
+    filename = re.sub(r"^gs://?", "", filename)
+    bucket_name, blob_name = filename.split("/", 1)
+
+    storage_client = storage.Client()
+    bucket = storage_client.bucket(bucket_name)
+    blob = bucket.blob(blob_name)
+
+    with blob.open("rb") as f:
+        return read_h5ad(f)
+
+
+def read_h5ad_filename(filename: str) -> AnnData:
```

Comment:
Consider further refactoring to `read_h5ad_local` and `read_h5ad` which handles mixed input.